torch/utils/data/datapipes/_typing.py

```diff
@@ -337,6 +337,32 @@ def _hash_(self):
         return hash((self.__name__, self.type))
 
 
+def simplify_value_name(value) -> str:
+    if inspect.isfunction(value):
+        return value.__name__
+    try:
+        return str(value.__class__.__qualname__)
+    except Exception as _:
+        return str(value)
+
+
+def get_input_args(obj):
+    signature = inspect.signature(obj.__class__)
+    input_param_names = set()
+    for param_name, _ in signature.parameters.items():
+        input_param_names.add(param_name)
+    result = []
+    for name, value in inspect.getmembers(obj):
+        if name in input_param_names:
+            result.append((name, simplify_value_name(value)))
+    return ', '.join([f'{name}={value}' for name, value in result])
+
+
+class DataPipeException(Exception):
```

Comment:
I am adding a custom Exception here so that when the exception propagates through the DataPipes, it will not need to be re-raised and can pass through.

Perhaps it can be helpful in cutting down the traceback as well? Any idea?

Reply:
Seems like starting with Python 3.7, we can use the `with_traceback` method. I will play around with that